fontir/src/ir.rs

```diff
@@ -1890,6 +1900,43 @@ pub struct Component {
     pub transform: Affine,
 }
 
+/// Data to inform construction of [CPAL](https://learn.microsoft.com/en-us/typography/opentype/spec/cpal#palette-table-header)
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq)]
+pub struct ColorPalettes {
+    // All palettes should have the same number of colors
+    pub palettes: Vec<Vec<Color>>,
+}
+
+impl ColorPalettes {
+    pub fn new(mut palettes: Vec<Vec<Color>>) -> Self {
```

Comment:
mentioned above, but it feels to me like this is an excellent place for us to go and do our validation:

- palettes isn't empty
- palettes all have the same length

Reply:
In the light of a new day I agree, so updated.